### PR TITLE
Changes to activate package only on Ruby file load

### PR DESCRIPTION
**Pull request by deiga**
This speeds up Atom startup

### CODE REVIEW

**Comment by ricardograca** [comment]
Any ideas why all the tests are failing? Seems to be related to this change.

**Comment by deiga** [comment]
Not right now, I'll check it out :)

**Comment by Arcanemagus** [comment]
When you switch to an `activationHooks` model the specs need to be modified to activate the package properly.

See [here](https://github.com/AtomLinter/linter-jscs/blob/v4.1.2/spec/linter-jscs-spec.js#L16-L28) for an example.

**Comment by deiga** [comment]
@ricardograca Now it should work :)

**Comment by ricardograca** [comment]
Great! Looks good to me, but I'm not that familiar with this activation stuff yet, so I'll leave the final word to @Arcanemagus.

File: spec/linter-erb-spec.js [hunk 1]
```diff
@@ -1,20 +1,38 @@
...
+      atom.packages.activatePackage('language-ruby'),
+    );
+
+    waitsForPromise(() =>
         atom.workspace.open(path.join(__dirname, 'fixtures', 'good.erb')),
-      );
-    });
+    );
+
+    atom.packages.triggerActivationHook('language-ruby:grammar-used');
```
Comment (Arcanemagus) [inline spec/linter-erb-spec.js]: The method I linked should work without this line, and as we already are reaching into the non-public API with the next line I'd rather this was removed.

(The opening of the `good.erb` file should have triggered this for us.)

Comment (deiga) [inline spec/linter-erb-spec.js]: @Arcanemagus I tested it, but it does not activate for some reason

Comment (deiga) [inline spec/linter-erb-spec.js]: Without this the tests time out when waiting for `linter-erb` since there is no `language-ruby` activation

Comment (Arcanemagus) [inline spec/linter-erb-spec.js]: If that's the case then the file you are opening isn't being recognized as Ruby.

Comment (deiga) [inline spec/linter-erb-spec.js]: The problem is that changing the code to be equal to `linter-jscs` example doesn't work, since the linter never activates

Comment (deiga) [inline spec/linter-erb-spec.js]: After some debugging it seems like the fixture file doesn't correctly instantiate the editor, as the scope for the file is `text.plain.null-grammar`

**Comment by deiga** [comment]
@ricardograca Fair enough, but with the correct file type it recognizes a grammar and with just `.erb` it doesn't. It might of course be an issue with `language-ruby` grammar definition, but I still would write the tests here so that everything passes and file an issue upstream

**Comment by deiga** [comment]
The problem here is, that I don't know how to debug Atom from this point onward. So I have no idea where the issue with plain `.erb` files is.

**Comment by ricardograca** [comment]
It's probably just a matter of adding the correct scope [in here](https://github.com/AtomLinter/linter-erb/blob/master/lib/index.js#L58). Have you tried that?

There's also the matter of the Travis build failing. It was also failing a little while ago, but it was already fixed. Maybe you branched off before that change. More info in #102.

**Comment by deiga** [comment]
I opened an upstream issue about this problem https://github.com/atom/language-ruby/issues/203

@ricardograca I can't fix this by adding a new scope as the `linter-erb` would work on all Plain Text files :)

The problem here is that ERB files actualyl have no grammar at all in Atom :) They used to all be `HTML (ERB)` files, but then somebody went and removed `.erb` from the grammars file extension. Now as `language-ruby-on-rails` has the file ext for `HTML (Rails)` Grammar it works in general, but tests where you have only limited packages makes it fail.

I could of course change the test to just activate `language-ruby-on-rails` instead of `language-ruby`, but it's still dancing around the issue of missing an actual grammar for ERB files

**Comment by deiga** [comment]
@ricardograca I did a rebase from master, but still Travis fails with a missing glibc

**Comment by ricardograca** [comment]
It seems a fix for that is on its way: https://github.com/atom/atom/issues/14827#issuecomment-309592418

**Comment by deiga** [comment]
@ricardograca Could you schedule a rebuild? I'd hate to push an empty commit just to restart travis :)

**Comment by ricardograca** [comment]
@deiga Done. It passed.

**Comment by deiga** [comment]
@Arcanemagus Is this now up to par, or should some changes still be applied?

**Comment by ricardograca** [comment]
Sorry it took so long and thanks for the contribution. We'll probably release a new version with these changes soon.

**Pull request merged**